File: lingting-core/src/main/java/live/lingting/framework/context/ContextHolder.java
package live.lingting.framework.context;

/**
 * @author lingting 2023-12-06 17:13
 */
public final class ContextHolder {

	private static boolean stop = true;

	private ContextHolder() {throw new UnsupportedOperationException("This is a utility class and cannot be instantiated");}

	public static void start() {
		stop = false;
	}

	public static void stop() {
		stop = true;
	}

	public static boolean isStop() {return ContextHolder.stop;}
}

File: lingting-core/src/main/kotlin/live/lingting/framework/context/ContextHolder.kt
package live.lingting.framework.context

/**
 * @author lingting 2023-12-06 17:13
 */
object ContextHolder {
    var isStop: Boolean = true
        private set

    fun start() {
        isStop = false
    }

    fun stop() {
        isStop = true
    }

}
